packages/client/src/workflow-client.ts

```diff
@@ -451,6 +457,33 @@ export interface IntoHistoriesOptions {
   bufferLimit?: number;
 }
 
+/**
+ * Defines how to start a workflow when using {@link WorkflowClient.startUpdateWithStart} and
+ * {@link WorkflowClient.executeUpdateWithStart}. `workflowIdConflictPolicy` is required in the options.
+ */
+export class WithStartWorkflowOperation<T extends Workflow> {
+  private _workflowHandle: Promise<WorkflowHandle<T>>;
+  private _resolveWorkflowHandle!: (handle: WorkflowHandle<T>) => void;
+  public _executed = false;
+
+  constructor(
+    public workflowTypeOrFunc: string | T,
+    public options: WorkflowStartOptions<T> & { workflowIdConflictPolicy: WorkflowIdConflictPolicy }
```

Comment:
Can live with this temporarily, but we should avoid ad-hoc types in public facing APIs.